docs/packaging-ios.md

```diff
@@ -0,0 +1,228 @@
+# Packaging React Native Modules as `.framework` for iOS
+
+React Native isn't limited to building standalone applications. Many developers aim to create reusable modules or components that can be incorporated into existing or new native iOS projects. A practical way to distribute and embed these modules is by packaging them as `.framework` files. This guide will walk you through packaging your React Native module as a `.framework` for iOS.
+
+## Why Package React Native?
+
+When incorporating React Native modules into a native app, it's frequently necessary to bring in JavaScript-specific tooling, such as `yarn`, `metro-bundler`, `jest`, and other related tools. However, this can create challenges, as it might raise the entry barrier for native development teams. Furthermore, it's not uncommon for React Native teams to operate independently from native teams. In situations like these, dividing development across separate repositories can be beneficial. This is where packaging becomes invaluable.
+
+## Key Concepts
+
+1. **.framework**: This is a directory structure in iOS and macOS development that packages a shared library along with necessary resources, such as headers, images, and localized strings. Essentially, it bundles related dynamic shared libraries and their associated resources into a single package.
+
+2. **Integrating with Existing Apps**: Packaging React Native Modules as a `.framework` file is often done to integrate with existing apps. Familiarize yourself with the [official guide on these integrations](https://reactnative.dev/docs/integration-with-existing-apps?language=objc).
+
+3. **Native Modules**: To allow native iOS to access React Native functionality, we use Native Modules. If you're unfamiliar with native modules, follow [this guide](https://reactnative.dev/docs/native-modules-ios).
+
+4. **Linking Dependencies**: It's crucial to link all dependencies of the React Native module accurately. This ensures that when the module is activated from a native app, it operates with all the required code and resources.
+
+## Prerequisites
+
+1. Follow the [standard native environment setup](https://reactnative.dev/docs/environment-setup?guide=native).
+
+## Creating a 2048 High Scores Package
+
+In this guide, you'll craft a High Scores screen, similar to the one made in the [guide to integrate with native apps](https://reactnative.dev/docs/integration-with-existing-apps?language=objc). You'll package it as a standalone `.xcframework` file. Afterward, you can use it directly in a native app like any native library, eliminating the need for installing any JavaScript dependencies.
+
+### Starting a New React Native App
+
+Begin by generating a fresh React Native project:
```

Comment:
Why do I need to create a whole project if it is said we can integrate components? I would like to see explanation if this is the only way.

Reply:
Right. In that case we can register multiple root views and duplicate the binding files. I think it would be good to explain that explicitly in the docs tho.